src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/Main.java

```diff
@@ -105,65 +108,115 @@ private static ReviewProperties getReviewProperties(File inputFile) {
-            final List<Path> allFiles = new ArrayList<>();
-            try (FileSystem fs = FileSystems.newFileSystem(inputFile.toPath(), Main.class.getClassLoader())) {
-                fs.getRootDirectories().forEach(root -> {
-                    try (Stream<Path> paths = Files.walk(root)) {
-                        paths.forEach(allFiles::add);
-                    } catch (IOException e) {
-                        e.printStackTrace();
-                        System.exit(-1);
-                    }
-                });
-
-                // Do the analysis while the filesystem is still represented in memory
-                analyser.analyse(allFiles);
+            // Do the analysis while the filesystem is still represented in memory
+            analyser.analyse(allFiles);
+        }
+    }
+
+    private static void processXmlFile(File inputFile, APIListing apiListing) {
+        final ReviewProperties reviewProperties = new ReviewProperties();
+
+        try (FileInputStream fis = new FileInputStream(inputFile)) {
+            String reviewName;
+            String packageName;
+            String packageVersion;
+
+            try {
+                reviewProperties.setMavenPom(new Pom(fis));
+
+                final String groupId = reviewProperties.getMavenPom().getGroupId();
+                packageName = (groupId.isEmpty() ? "" : groupId + ":") + reviewProperties.getMavenPom().getArtifactId();
+                packageVersion = reviewProperties.getMavenPom().getVersion();
+
+                reviewName = reviewProperties.getMavenPom().getArtifactId() + " (version " + packageVersion + ")";
+
+                apiListing.setMavenPom(reviewProperties.getMavenPom());
+            } catch (IOException e) {
+                // this is likely to not be a maven pom file
+                reviewName = inputFile.getName();
+                packageName = reviewName;
+                packageVersion = "1.0.0";
             }
 
+            System.out.println("  Using '" + reviewName + "' for the review name");
+            System.out.println("  Using '" + packageName + "' for the package name");
+            System.out.println("  Using '" + packageVersion + "' for the package version");
 
-        } else {
-            apiListing.getTokens().add(new Token(LINE_ID_MARKER, "Error!", "error"));
-            apiListing.addDiagnostic(new Diagnostic(
-                DiagnosticKind.ERROR,
-                "error",
-                "Uploaded files should end with '-sources.jar', " +
-                    "as the APIView tool only works with source jar files, not compiled jar files. The uploaded file " +
-                    "that was submitted to APIView was named " + inputFile.getName()));
+            apiListing.setReviewName(reviewName);
+            apiListing.setPackageName(packageName);
+            apiListing.setPackageVersion(packageVersion);
+            apiListing.setLanguage("Xml");
```

Comment:
Because we support any XML, and not all of them will necessarily be pom, I wanted a generic location. I could do it dynamically and put pom under Java, but that might be more confusing... Thoughts?

Reply:
Yeah, if line 188 succeeds without exception, we can mark the language as `Java` otherwise use `Xml`.

Reply:
Right, it's possible, but is it the right thing to do?

Reply:
If we could categorize it as both Java and Xml that would have been great, so, it would show up under both tabs. But I would expect BOM files to show up when I click on the Java tab. I guess it's a minor issue and not a deal breaker if it was all under "Xml" either.

Reply:
I'll put poms under java for now then.